bcgov/GDX-Analytics-microservice
# Gdxdsd 7197 google mybusiness servicebc derived split

### This PR does the following:

1. Moves and/or copies the python scripts google_mybusiness_servicebc_derived.py, along with its associated config, ddl, and development files into a separate google_mybusiness_servicebc_derived folder
2. Creates new python virtual environment files that only use packages specific to google_mybusiness_servicebc_derived.py
3. Edits the README.md file to removed details specific to the google_search, and google_mybusiness microservices

Please note that some files have been modified for testing purposes. You can view the modified files in the ticket

### Login to the development environment:

1. Run the following command to login to EC2
```
$ awsmfa prod ######
Generating new IAM STS Token ...
STS Session Token generated and saved in profile mfa successfully.
$ microservice_ssm
```
2. Change the directory to the development branch:
```
$ cd /home/microservice/branch/GDXDSD-7197_google_mybusiness_servicebc_derived_split/google_mybusiness_servicebc_derived
```

### Review README.md
Review [README.md](https://github.com/bcgov/GDX-Analytics-microservice/blob/GDXDSD-7197_google_mybusiness_servicebc_derived_split/google_mybusiness_servicebc_derived/README.md) and check to makes sure there are no details specific to google_search and google_mybusiness. Also make sure the details about google_mybusiness_servicebc_derived.py makes sense. 
Please note that there are possibly still references to these other microservices where relevant. But these references shouldn't be related to how the microservice works or how the configs are structured

### Review moved files

Part of this changes includes moving files from the old combined folder into a separate one specific for this microservice. In this section, we will be reviewing to make sure all of these files were moved successfully
1. Review [google_mybusiness_servicebc_derived](https://github.com/bcgov/GDX-Analytics-microservice/tree/GDXDSD-7197_google_mybusiness_servicebc_derived_split/google_mybusiness_servicebc_derived) to make sure the microservice scripts have been moved successfully
- google_mybusiness_servicebc_derived.py
2. Run the following command to get a list of the scheduled microservice's configs:
```
$ crontab -l | grep google_mybusiness_servicebc_derived
```
4. Review the files in [config.d](https://github.com/bcgov/GDX-Analytics-microservice/tree/GDXDSD-7197_google_mybusiness_servicebc_derived_split/google_mybusiness_servicebc_derived/config.d) and make sure any config that appeared in the previous step is listed
5. Review the files in [dml](https://github.com/bcgov/GDX-Analytics-microservice/tree/GDXDSD-7197_google_mybusiness_servicebc_derived_split/google_mybusiness_servicebc_derived/dml) and make sure any file listed in one of the previous step's configs has an associated file here.

### Review copied files

1. Review the files in [google_mybusiness_servicebc_derived](https://github.com/bcgov/GDX-Analytics-microservice/tree/GDXDSD-7197_google_mybusiness_servicebc_derived_split/google_mybusiness_servicebc_derived) to make sure the following file exist
- .flake8
- pylintrc
- README.md

### Review created files
1. Review the files in [google_mybusiness_servicebc_derived](https://github.com/bcgov/GDX-Analytics-microservice/tree/GDXDSD-7197_google_mybusiness_servicebc_derived_split/google_mybusiness_servicebc_derived) to make sure the following file exist
- Pipfile
- Pipfile.lock

### Testing the microservice:
1. Run the following command to test the google_mybusiness.py script and compare it to the expected output:
```
$ /home/microservice/.local/bin/pipenv run python3.8 google_mybusiness_servicebc_derived.py -c config.d/GDXDSD-7197_config_servicebc.json
Success: executed the transaction to prepare the google_mybusiness_servicebc_derived DT
```
2. Run the following query to check if the data was loaded correctly:
```
$ microservice_rs
SET

snowplow=> select count(*) from test.gdxdsd_7197_google_mybusiness_servicebc_derived;
 count 
-------
   109
(1 row)

snowplow=> \q
```
3. Run the following queries to drop the test tables:
```
$ microservice_rs
SET

snowplow=> drop table test.gdxdsd_7197_google_mybusiness_servicebc_derived;
DROP TABLE
snowplow=> \q
```
